lib/queue.js

```diff
@@ -1,27 +1,67 @@
+
+
 class Queue {
-    constructor() {
-        // this.store = ...
-        throw new Error("This method has not been implemented!");
+    constructor(size = 25) {
+        this.size = size
+        this.store = new Array(size);
+        this.front = null;
+        this.rear = null;
     }
 
     enqueue(element) {
-        throw new Error("This method has not been implemented!");
+        if (this.front == 0 && (this.rear == this.store.size - 1)){
+            return false
```

Comment:
This `if` condition will prevent the rear from wrapping around the buffer, which is the behavior you want. 

I think it should instead be:

```suggestion
        if ((this.rear +1) % this.store.size == this.front){
            return false
```